src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -722,11 +722,9 @@ impl<'a, 'tcx> Visitor<'tcx> for Qualifier<'a, 'tcx, 'tcx> {
             }
 
             Rvalue::Discriminant(..) => {
-                // FIXME discriminant
+                // FIXME implement discriminant const qualify
                 self.add(Qualif::NOT_CONST);
-                if self.mode != Mode::Fn {
-                    bug!("implement discriminant const qualify");
-                }
+                // Discriminants in consts will error elsewhere as an unimplemented expression type
```

Comment:
You are not calling `.not_const()` because we report an error for the `SwitchInt` terminator too?